src/components/CheckIn.tsx

```diff
@@ -1,12 +1,21 @@
 import { TableCell, TableRow, Typography } from "@mui/material"
-import { PropsWithChildren } from "react"
-import { CheckIn } from "../models/checkin"
+import axios from "axios"
+import { PropsWithChildren, useEffect, useState } from "react"
+import { CheckIn, Event } from "../models/checkin"
+import { get_all_checkins_by_id } from "../services/CheckIn"
+
+export const CheckInRow = ({ checkin }: PropsWithChildren<{ checkin: CheckIn }>) => {   
+  
+  const get_all_events = async () => {
+      let all_events: Array<Event> = (await axios.get("http://localhost:3001/v1/event")).data;
+      const match_events_to_checkin = all_events.filter((event) => event.id == checkin.event)
+      return match_events_to_checkin[0].alias;
+  }
 
-export const CheckInRow = ({ checkin }: PropsWithChildren<{ checkin: CheckIn }>) => {
   return (
       <TableRow>
         <TableCell>
-
+          <Typography>{get_all_events().toString()}</Typography>
```

Comment:
2. Let's think about what `get_all_events` is returning. It is an `async` function, and `async` returns a `Promise<T>`. Problem is, we have no possible way of displaying a `Promise<T>`.  Instead, we need to set a state variable to the result given inside of `get_all_events` (i.e. set the state instead of returning a value). Then, we use the *state variable* in the HTML rather than the function.
This ALSO means we have to call `get_all_events` in an `useEffect(() => get_all_events(), [])`

Remember how we used this pattern before?